### PR TITLE
Send token in Authorization header

### PR DESCRIPTION
**Pull request by CalvinScott**
We are currently appending access_token as a parameter in the request method. Good practice dictates the passing of bearer tokens via `Authorization: Bearer ...` in the request headers.

### CODE REVIEW

**Comment by CalvinScott** [comment]
@slajax @switters @lvivier @RongxinZhang Anyone free to review?

File: index.js [hunk 1]
```diff
@@ -287,12 +287,13 @@ Retsly.prototype.request = function(method, url, query, cb) {
 
   options.method = method;
   options.url = url;
-  options.query.client_id = this.client_id;
+ //options.query.client_id = this.client_id;
```
Comment (slajax) [inline index.js]: you can delete this

File: index.js [hunk 2]
```diff
@@ -287,12 +287,13 @@ Retsly.prototype.request = function(method, url, query, cb) {
...
 
-  if(this.getUserToken())
-    options.query.access_token = this.getUserToken();
+ var token;
+ if(this.getUserToken())
+    token = this.getUserToken();
   else if(this.getAppToken())
```
Comment (Jkhwan) [inline index.js]: some spacing issue here

**Comment by Jkhwan** [comment]
also curious to know why the passport reorder branch was required? Does it not work if they are not reordered?

**Comment by CalvinScott** [comment]
Apparently some have precedence when they are ordered in different ways, and cause constant logout refreshing

**Comment by Jkhwan** [comment]
@CalvinScott the order should not matter. I tested it out, even with the reorder, once you refresh the page on www, you will be logged out. The underlying problem is with the api/v1/user/me endpoint. It is expecting to find the token in req.query.access_token (www is using retsly-js-sdk) for the request, when token is not found, it threw an authorization error, thus logging the user out.

**Comment by CalvinScott** [comment]
@Jkhwan Ah I see, I think it will have to be something I look into more tomorrow. Kyle was running over most of it with me since I didn't fully understand what was going on and we noticed that it was getting stuck (and fixed) when they were ordered differently. 

I think some of the issues in this pull-request may be a bit too deep for me yet :dizzy_face:

**Comment by Jkhwan** [comment]
No worries. I didn't mean to jump all over this issue, I was just curious to test it out because I never knew that the order of the passport strategies would matter in this instance. Besides, most of the passport codes are being moved out to authd anyways.

**Comment by slajax** [comment]
@Jkhwan Thanks for jumping in man. It seemed like the order matter slightly because there is a bit of an overlap between the `apiKey` and `clientPassword` strategies I think. In either case, reordering it seemed to help. If you have a spare cycle or two tomorrow, maybe you can help Cal debug this?

**Comment by CalvinScott** [comment]
After some more testing (24 test cases for all of the different patterns) it turns out that the ordering ultimately doesn't matter. To be honest I don't know where the weird refresh error that Kyle and I were seeing came from. Because the ordering is arbitrary, I will leave them as they are in this PR for simplicity (along with Kyle's new comments there), unless anyone has a problem with that.

**Comment by CalvinScott** [comment]
Stefan isn't around so I'm not sure who else to tag for review. Any of you open for a quick check :sweat_smile: ?  @CMClay @RongxinZhang @Jkhwan @MadManMathew

**Comment by Jkhwan** [comment]
:+1:

**Comment by slajax** [comment]
Good job with the thorough testing @CalvinScott ! :+1: You can merge it in.

**Comment by slajax** [comment]
Grab me when you are ready and I'll show you how to tag the release so you can bump deps in www.

**Pull request merged**